src/Service.php

```diff
@@ -7,179 +8,315 @@
-                    'nickname'         => $method . ' for ' . $service->getName(),
-                    'type'             => $service->getName(),
-                    'parameters'       => $entityParameters,
-                    'responseMessages' => $entityOperation->getResponseStatusCodes(),
-                ];
+    protected function getModelFromFields()
+    {
+        $required = $properties = [];
+        $fields = $this->getFieldsForDefinitions();
+        foreach ($fields as $field) {
+            $properties[$field->getName()] = $this->getFieldProperties($field);
+            if ($field->isRequired()) {
+                $required[] = $field->getName();
             }
+        }
+        return $this->cleanEmptyValues([
+                'type' => 'object',
+                'properties' => $properties,
+                'required' => $required
+        ]);
+    }
 
-            $operationGroups[] = [
-                'operations' => $collectionOperations,
-                'path'       => str_replace('/{' . $service->routeIdentifierName . '}', '', $routeWithReplacements)
-            ];
+    protected function getModelFromFirstPostDescription()
+    {
+        $firstPostDescription = $this->getFirstPostRequestDescription();
+        $model = $this->modelGenerator->generate($firstPostDescription);
+        return $model ? $model : [];
+    }
 
-            $operationGroups[] = [
-                'operations' => $entityOperations,
-                'path' => $routeWithReplacements
-            ];
-        } else {
-            // find all other operations
-            $operations = [];
-            foreach ($service->operations as $operation) {
-                $method     = $operation->getHttpMethod();
-                $parameters = array_values($templateParameters);
-
-                if (in_array($method, ['POST', 'PUT', 'PATCH'])) {
-                    $parameters[] = $postPatchPutBodyParameter;
-                }
-
-                $operations[] = [
-                    'method'           => $method,
-                    'summary'          => $operation->getDescription(),
-                    'notes'            => $operation->getDescription(),
-                    'nickname'         => $method . ' for ' . $service->getName(),
-                    'type'             => $service->getName(),
-                    'parameters'       => $parameters,
-                    'responseMessages' => $operation->getResponseStatusCodes(),
-                ];
+    protected function getFirstPostRequestDescription()
+    {
+        foreach ($this->getAllOperations() as $operation) {
+            $method = $this->getMethodFromOperation($operation);
+            if ($method === 'post') {
+                return $operation->getRequestDescription();
             }
-            $operationGroups[] = [
-                'operations' => $operations,
-                'path'       => $routeWithReplacements
-            ];
         }
```

Comment:
Is this method expected to return `void`/`null` if no `post` operations are found?